hw_diag/diagnostics/key_diagnostics.py

```diff
@@ -25,8 +24,12 @@ def perform_test(self, diagnostics_report):
         # Try to get key, but there may be ECC lock or other failure
         try:
             public_keys = get_public_keys_rust()
+        except UnboundLocalError as e:
```

Comment:
Is there a reason that the exceptions that we are catching here are different than the ones for `EccDiagnostic`? Both `get_gateway_mfr_result` and `get_public_keys_rust` invoke `run_gateway_mfr` so they should have the same exceptions.

It doesn't really matter since there is a catch-all `except Exception as e` in both cases.

Reply:
Yeah so the reason this one has UnboundLocal is that I experienced it directly when testing this, I could add it to the rest but like you said we are catching all exceptions just in case one falls through that we didn't put a specific catch case for. In this specific scenario, I will add UnboundLocal to all of them it's just that since get_public_keys is relying on the rest there is a chance that if something fails it won't be bound so in this specific try catch you marked above it makes sense to include it to prevent running the rest in case it's unbound.

Reply:
To be clear, I'm just talking about UnboundLocal. `get_public_keys_rust` can also potentially throw ECCMalfunctionException,  GatewayMFRFileNotFoundException, and ResourceBusyError. I'm also not sure it can still throw FileNotFoundError, NotADirectoryError as those are [now being caught](https://github.com/NebraLtd/hm-pyhelper/blob/95676fa2c05b8b8d7cb307313f91eb3353f7a653/hm_pyhelper/miner_param.py#L51) in hm-pyhelper.

Reply:
> To be clear, I'm not just talking about UnboundLocal. `get_public_keys_rust` can also potentially throw ECCMalfunctionException, GatewayMFRFileNotFoundException, and ResourceBusyError. I'm also not sure it can still throw FileNotFoundError, NotADirectoryError as those are [now being caught](https://github.com/NebraLtd/hm-pyhelper/blob/95676fa2c05b8b8d7cb307313f91eb3353f7a653/hm_pyhelper/miner_param.py#L51) in hm-pyhelper.

Oh ok, I'll take a look at it and see what's being caught and what I can add, I just did a push with some changes let me know if they make sense. I'm currently looking at the changes and reviewing to see if Unit Tests can be added and I'll take a look at the functions you mentioned above and see what else I can catch

Reply:
I'm now catching these in KeyDiagnostics the rest of it is all in hm_pyhelper the following functions are present there
`get_gateway_mfr_result` `get_public_keys_rust``run_gateway_mfr`